src/js/event-target.js

```diff
@@ -11,7 +11,7 @@ EventTarget.prototype.on = function(type, fn) {
   // Remove the addEventListener alias before calling Events.on
   // so we don't get into an infinite type loop
   let ael = this.addEventListener;
-  this.addEventListener = Function.prototype;
+  this.addEventListener = () => {};
```

Comment:
That's the crux of the discussion I think we should have. Using fat-arrows removes the ability of the caller to rebind `this`. I've felt it necessary to rebind `this` more than once in my life. Are we intentionally removing that ability or just doing it to save some keystrokes?